### PR TITLE
circe -> 0.14.3, scala to latest, other upgrades

### PR DESCRIPTION
**Pull request by barryoneill**

### CODE REVIEW

**Comment by zarthross** [comment]
Thanks for the PR @barryoneill.  Can you update to latest master and run `sbt githubWorkflowGenerate` please and thank you.

**Comment by barryoneill** [comment]
@zarthross done!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,19 +22,21 @@ jobs:
...
 
-      - name: Setup Java and Scala
-        uses: olafurpg/setup-scala@v13
+      - name: Setup Java (adopt-hotspot@1.8)
+        if: matrix.java == 'adopt-hotspot@1.8'
+        uses: actions/setup-java@v2
         with:
-          java-version: ${{ matrix.java }}
+          distribution: adopt-hotspot
+          java-version: 8
```
Comment (barryoneill) [inline .github/workflows/ci.yml:39]: [Could not find satisfied version for SemVer '1.8'](https://github.com/actions/setup-java/blob/main/docs/switching-to-v2.md#dropping-legacy-java-version-syntax-1x)

**Comment by barryoneill** [comment]
@zarthross Hi - can you trigger this again?  I'd like to get this in before it goes out of date again :).  (thank you in advance)

**Comment by barryoneill** [comment]
image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zarthross) [reply .github/workflows/ci.yml]: `sbt githubWorkflowGenerate`

Comment (zarthross) [reply .github/workflows/ci.yml]: should be what you run....

**Comment by barryoneill** [comment]
Thanks.  Will be interesting to see if this works, because the config that the command generated previously was causing the build to fail.

**Comment by zarthross** [comment]
@barryoneill Update everything bug the github actions, and don't worry about it.  We are working on using sbt-circe-org plugin on all the repos (eventually) and it'll take care of the github actions.

**Comment by barryoneill** [comment]
this is as up to date as I can make it :)

**Comment by zarthross** [comment]
Sorry, typo, `update every EXCEPT the github actions`

**Comment by barryoneill** [comment]
I'm a little confused.  My understanding is:
* Per instructed, I ran `sbt githubWorkflowGenerate`  ( https://github.com/circe/circe-optics/pull/286/commits/927fa8a3d5bb97d69afef429dbad8129aec15478 )
* That failed, so I attempted to fix the flow ( https://github.com/circe/circe-optics/pull/286/commits/d2f2097a054dff54d3ab3b8a8d8e7c333b26c616 ) 
* That failed, and you pushed the workflow generation, which effectively backed out my change ( https://github.com/circe/circe-optics/pull/286/commits/b129288b2f0a621c45577902f527189a2ffcb404 ) 

Everything except the actions are as up to date as I can make them...

**Comment by barryoneill** [comment]
I could revert https://github.com/circe/circe-optics/commit/927fa8a3d5bb97d69afef429dbad8129aec15478  ...

**Comment by zmccoy** [comment]
Thanks for your work, this was superceded by the update to the circe org plugin.

**Pull request closed without merging**